Installing on kubernetes 1.21.2 gives error on fresh install due to usage of policy/v1beta1

Using helm install on prometheus-community/kube-prometheus-stack on kubernetes 1.21.2 gives many warnings. In our case we do this using Azure DevOps build and the log has many warnings like this:

`W0930 08:26:58.656126    1843 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+`

Eventually it gives an error on this warning:

`##[error]W0930 08:26:47.154610    1843 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+`

The same pipeline using latest version of prometheus-community/kube-prometheus-stack on kubernetes version 1.20.5 works OK. 

EDIT: To clarify, the installation also is a success, but helm writes these errors to **stderr** and the build will indicate failure.

Issue:
policy/v1beta1 will be deprecated in favor of policy/v1. Even though it's supposed to be just a warning until 1.25, it throws an error event stopping the installation of prometheus.

Suggested solution:
This kind of logic for policy version selection:
https://github.com/prometheus-community/helm-charts/blob/383b15a4c60f8cc33975f90d8e705e72052d6713/charts/prometheus/templates/_helpers.tpl#L187-L195

--- a/charts/kube-prometheus-stack/Chart.yaml
+++ b/charts/kube-prometheus-stack/Chart.yaml
@@ -17,7 +17,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 27.1.0
+version: 28.0.0
 appVersion: 0.53.1
 kubeVersion: ">=1.16.0-0"
 home: https://github.com/prometheus-operator/kube-prometheus

--- a/charts/kube-prometheus-stack/README.md
+++ b/charts/kube-prometheus-stack/README.md
@@ -83,6 +83,12 @@ _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documen
 
 A major chart version change (like v1.2.3 -> v2.0.0) indicates that there is an incompatible breaking change needing manual actions.
 
+### From 27.x to 28.x
+
+This version disables PodSecurityPolicies by default because they are deprecated in Kubernetes 1.21 and will be removed in Kubernetes 1.25.
+
+If you are using PodSecurityPolicies you can enable the previous behaviour by setting `kube-state-metrics.podSecurityPolicy.enabled`, `prometheus-node-exporter.rbac.pspEnabled`, `grafana.rbac.pspEnabled` and `global.rbac.pspEnabled` to `true`.
+
 ### From 26.x to 27.x
 
 This version splits Node Exporter recording and altering rules in separate config values.

--- a/charts/kube-prometheus-stack/values.yaml
+++ b/charts/kube-prometheus-stack/values.yaml
@@ -101,7 +101,7 @@ additionalPrometheusRulesMap: {}
 global:
   rbac:
     create: true
-    pspEnabled: true
+    pspEnabled: false
     pspAnnotations: {}
       ## Specify pod annotations
       ## Ref: https://kubernetes.io/docs/concepts/policy/pod-security-policy/#apparmor
@@ -649,6 +649,11 @@ grafana:
   defaultDashboardsTimezone: utc
 
   adminPassword: prom-operator
+
+  rbac:
+    ## If true, Grafana PSPs will be created
+    ##
+    pspEnabled: false
 
   ingress:
     ## If true, Grafana Ingress will be created
@@ -1306,8 +1311,6 @@ kube-state-metrics:
   namespaceOverride: ""
   rbac:
     create: true
-  podSecurityPolicy:
-    enabled: true
   prometheus:
     monitor:
       enabled: true
@@ -1406,6 +1409,10 @@ prometheus-node-exporter:
       #   targetLabel: nodename
       #   replacement: $1
       #   action: replace
+  rbac:
+    ## If true, create PSPs for node-exporter
+    ##
+    pspEnabled: false
 
 ## Manages Prometheus and Alertmanager components
 ##

